src/borg/fuse.py

```diff
@@ -339,12 +341,19 @@ def process_archive(self, archive_name, prefix=[]):
         logger.debug('fuse: process_archive completed in %.1f s for archive %s', duration, archive.name)
 
     def process_leaf(self, name, item, parent, prefix, is_dir, item_inode):
-        def file_version(item):
+        def file_version(item, path):
             if 'chunks' in item:
-                ident = 0
-                for chunkid, _, _ in item.chunks:
-                    ident = adler32(chunkid, ident)
-                return ident
+                file_id = blake2b_128(path)
+                current_version, previous_id = self.versions_index.get(file_id, (0, None))
+
+                chunk_ids = [chunk_id for chunk_id, _, _ in item.chunks]
+                contents_id = blake2b_128(b''.join(chunk_ids))
+
+                if contents_id != previous_id:
```

Comment:
putting (potentially long) archive names into the filename seems weird.

but we could build a structure like this:
```
path/filename.ext/filename.N.ext --> ../.../archivename/path/filename.ext
```
- less ugly filenames
- same "issue" with showing all version
- added benefit to be able to find into the corresponding archive via filesystem.

although using a FILE symlink is maybe not most comfortable, so maybe we would have additionally a symlink to the parent directory in the respective archive.